opm/simulators/utils/readDeck.cpp

```diff
@@ -230,6 +237,24 @@ namespace {
 #endif
     }
 
+    void inconsistentScheduleError(const std::string& message)
+    {
+        OPM_THROW(std::logic_error,
+                  fmt::format("Inconsistent SCHEDULE section: {}", message));
+    }
+
+    void checkScheduleKeywordConsistency(const Opm::Schedule& schedule)
+    {
+        const auto& final_state = schedule.back();
+        const auto& rescoup = final_state.rescoup();
+        if (rescoup.slaveCount() > 0 && rescoup.masterGroupCount() == 0) {
+            inconsistentScheduleError("SLAVES keyword without GRUPMAST keyword");
+        }
+        if (rescoup.slaveCount() == 0 && rescoup.masterGroupCount() > 0) {
+            inconsistentScheduleError("GRUPMAST keyword without SLAVES keyword");
+        }
```

Comment:
> What happens if we have SLAVES and GRUPMAST but at different dates/steps?

@blattms Good point. I initially thought that it would only make sense to have them at the same step at the beginning of the schedule. But then they did not really belong in the SCHEDULE section, so it seemed a little bit strange. Now I can see that it can make sense to have them at another step than report step 0 if it is a restart run. I need to fix my assumption [here](https://github.com/OPM/opm-common/pull/4123/files#diff-ce20b55c5a4cc52aeeec620d4579533d0cc760f8fd27a5da4fa8a1fd96a40790R63) then. 

Anyway, `GRUPMAST` can be allowed to occur before `SLAVES` but not the other way around since the `SLAVES` keyword will start the slave processes. Also, it is an error to have more than one occurrence of `SLAVES` in the schedule.

> Shouldn't we rather do this check during parsing in opm-common?

We can check if `GRUPMAST` is missing when `SLAVES` is encountered inside the keyword handlers in opm-common. However, it is not clear to me how to check if `SLAVES` is missing inside the handlers. That might be easier to do after the Schedule has been created in opm-simulators.